src/command_modules/azure-cli-kusto/azure/cli/command_modules/kusto/custom.py

```diff
@@ -0,0 +1,173 @@
+
+    return _client.start(resource_group_name=resource_group_name,
+                         cluster_name=cluster_name,
+                         custom_headers=custom_headers,
+                         raw=raw,
+                         polling=polling,
+                         operation_config=kwargs)
+
+
+def cluster_stop(cmd,
+                 resource_group_name,
+                 cluster_name,
+                 custom_headers=None,
+                 raw=False,
+                 polling=True,
+                 **kwargs):
+
+    from azure.cli.command_modules.kusto._client_factory import cf_cluster
+
+    _client = cf_cluster(cmd.cli_ctx, None)
+
+    return _client.stop(resource_group_name=resource_group_name,
+                        cluster_name=cluster_name,
+                        custom_headers=custom_headers,
+                        raw=raw,
+                        polling=polling,
+                        operation_config=kwargs)
+
+
+def database_create(cmd,
+                    resource_group_name,
+                    cluster_name,
+                    database_name,
+                    soft_delete_period_in_days,
+                    hot_cache_period_in_days=None,
+                    custom_headers=None,
+                    raw=False,
+                    polling=True,
+                    no_wait=False,
+                    **kwargs):
+
+    from azure.mgmt.kusto.models import Database
+    from azure.cli.command_modules.kusto._client_factory import cf_database
+
+    _client = cf_database(cmd.cli_ctx, None)
+    _cluster = _cluster_get(cmd, resource_group_name, cluster_name, custom_headers, raw, **kwargs)
+
+    if no_wait:
+        location = _cluster.output.location
+    else:
+        location = _cluster.location
+
+    _database = Database(location=location,
+                         soft_delete_period_in_days=soft_delete_period_in_days,
+                         hot_cache_period_in_days=hot_cache_period_in_days)
+
+    return sdk_no_wait(no_wait,
+                       _client.create_or_update,
+                       resource_group_name=resource_group_name,
+                       cluster_name=cluster_name,
+                       database_name=database_name,
+                       parameters=_database,
+                       custom_headers=custom_headers,
+                       raw=raw,
+                       polling=polling,
+                       operation_config=kwargs)
+
+
+def update_kusto_cluster(instance, sku_name=None, capacity=None):
+    """
+    Update sku kusto cluster.
+
+    :param sku_name: the name of the sku.
+    """
+    from azure.mgmt.kusto.models import AzureSku
+    if sku_name is None:
+        sku_name = instance.sku.name
+    if capacity is None:
+        capacity = instance.sku.capacity
+    instance.sku = AzureSku(name=sku_name, capacity=capacity)
+    return instance
+
+
+def update_kusto_database(instance, soft_delete_period_in_days, hot_cache_period_in_days=None):
```

Comment:
You need to alias these to `--soft-delete-period` and `--hot-cache-period` as we don't allow units in parameter names. You can either add the alias in params.py or change the variable names here accordingly.